lib/node_modules/@stdlib/math/stats/kde2d/README.md

```diff
@@ -0,0 +1,180 @@
+<!--
+
+@license Apache-2.0
+
+Copyright (c) 2018 The Stdlib Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# 2 Dimensional Normal Kernel Density Estimation
+
+> Two-Dimensional KDE
+
+<section class="usage">
+
+## Usage
+
+```javascript
+var kde2d = require( '@stdlib/math/stats/kde2d' );
+```
+
+### kde2d( x, y\[, opts] )
+
+By default, the function computes two-dimensional normal kernel density estimation for data provided in [arrays][mdn-array] or [typed-arrays][mdn-typed-array] `x` and `y`.
+
+```javascript
+// Data from a normal:
+var x = [ 0.6333, 0.8643, 1.0952, 1.3262, 1.5571, 1.7881,
+    2.019, 2.25, 2.481, 2.7119 ];
+var y = [ -0.0468, 0.8012, 1.6492, 2.4973, 3.3454,
+    4.1934, 5.0415, 5.8896, 6.7376, 7.5857 ];
+
+var out = kde2d( x, y );
+/* e.g., returns
+    {
+        'x': [0.6333, 0.7199083333333333, ...],
+        'y': [-0.0468, 0.27122083333333336, ...],
+        'z': ndarray { Float64Array [0.04547178438418015, ...]}
+    }
+*/
+```
+
+The function accepts the following `options`:
+
+-   **h**: `NumberArray` of length 2 indicating the X and Y bandwidth values, respectively.
+-   **n**: A positive `integer` indicating the number of partitions to create in the grid. Default: `25`.
+-   **xLim**: An `array` of length 2 indicating the lower and upper limits of X to be considered when calculating the grid. The first element must be strictly less than the second element.
+-   **yLim**: An `array` of length 2 indicating the lower and upper limits of Y to be considered when calculating the grid. The first element must be strictly less than the second element.
+
+By default, the bandwidth argument is set by a builtin function. To choose different bandwidth values, set the `h` option. Note that if you use a custom bandwidth for one axis, you must also use a custom bandwidth for a second axis.
+
+```javascript
+var x = [ 0.6333, 0.8643, 1.0952, 1.3262, 1.5571, 1.7881,
+    2.019, 2.25, 2.481, 2.7119 ];
+var y = [ -0.0468, 0.8012, 1.6492, 2.4973, 3.3454, 4.1934,
+    5.0415, 5.8896, 6.7376, 7.5857 ];
+
+var out = kde2d( x, y, {
+    'h': [0.05, 0.1]
+} );
+/* e.g., returns
+    {
+        'x': [0.148, 0.3772, ...],
+        'y': [-1.1511, -0.253, ...],
+        'z': ndarray { Float64Array [6.344e-154, 1.93e-171, ...]}
```

Comment:
Ah, that's interesting: @kgryte, do we have precedent on how `ndarray`s ought to be displayed in return annotations? Any suggestions?

Reply:
I'd consult the `ndarray` docs: https://github.com/stdlib-js/stdlib/tree/369e44cb9f62e8f26e290158f8d39aab16596b3f/lib/node_modules/%40stdlib/ndarray/ctor#ndarray-buffer-shape-strides-offset-order-

In general, I'd opt for not displaying the contents of an ndarray, but, for a string representation, this is the current format: https://github.com/stdlib-js/stdlib/tree/369e44cb9f62e8f26e290158f8d39aab16596b3f/lib/node_modules/%40stdlib/ndarray/ctor#ndarrayprototypetostring . This is, however, subject to change.